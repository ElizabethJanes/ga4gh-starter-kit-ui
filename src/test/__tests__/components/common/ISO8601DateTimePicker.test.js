import React from 'react';
import renderer from 'react-test-renderer';
import ISO8601DateTimePicker from '../../../../lib/components/common/ISO8601DateTimePicker';

//will need to use mock functions to test different timezones, morning, evening, current time, etc.
test('SHOW <ISO8601DateTimePicker /> should handle any date and time', () => {
    const iso8601DateTimePicker = renderer.create(<ISO8601DateTimePicker 
        readOnly={true}
        value='2021-07-22T13:00:00Z'
        propertyName='Date-Time Snapshot Test'
        label='Date-Time Snapshot Test'
        description='This is a date-time field description for a snapshot test.' />)
    .toJSON();
    //expect(iso8601DateTimePicker).toMatchSnapshot();
})

test('NEW and EDIT <ISO8601DateTimePicker /> should handle any date and time', () => {
    const iso8601DateTimePicker = renderer.create(<ISO8601DateTimePicker 
        readOnly={false}
        value='2021-07-22T13:00:00Z'
        propertyName='Date-Time Snapshot Test'
        label='Date-Time Snapshot Test'
        description='This is a date-time field description for a snapshot test.' />)
    .toJSON();
    //expect(iso8601DateTimePicker).toMatchSnapshot();
})