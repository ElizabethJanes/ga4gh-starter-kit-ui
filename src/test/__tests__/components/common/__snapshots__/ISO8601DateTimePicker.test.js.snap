// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NEW and EDIT <ISO8601DateTimePicker /> should handle any date and time in EDT timezone 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    format="yyyy-MM-dd HH:mm:ss OOOO"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      for="test_date_time_picker"
      id="test_date_time_picker-label"
    >
      Test Date Time Picker
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-describedby="test_date_time_picker-helper-text"
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        id="test_date_time_picker"
        name="test_date_time_picker"
        readonly=""
        type="text"
        value="2021-07-22 09:00:00 GMT-04:00"
      />
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-filled"
      id="test_date_time_picker-helper-text"
    >
      This is a description of a date time picker.
    </p>
  </div>
</div>
`;

exports[`SHOW <ISO8601DateTimePicker /> should handle any date and time in EDT timezone 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    format="yyyy-MM-dd HH:mm:ss OOOO"
    readonly=""
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      for="test_date_time_picker"
      id="test_date_time_picker-label"
    >
      Test Date Time Picker
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-describedby="test_date_time_picker-helper-text"
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        id="test_date_time_picker"
        name="test_date_time_picker"
        readonly=""
        type="text"
        value="2021-07-22 09:00:00 GMT-04:00"
      />
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-filled"
      id="test_date_time_picker-helper-text"
    >
      This is a description of a date time picker.
    </p>
  </div>
</div>
`;
